.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,54 @@
+name: Run Pytest Tests
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Clean unused files
+        run: |
+          sudo rm -rf /usr/local/lib/android # will release about 10 GB if you don't need Android
+          sudo rm -rf /usr/share/dotnet # will release about 20GB if you don't need .NET
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Restore Docker layers cache
+        id: restore-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.ref_name }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Build Docker image
+        run: |
+          docker buildx build --cache-from=type=local,src=/tmp/.buildx-cache \
+                              --cache-to=type=local,dest=/tmp/.buildx-cache-new,mode=max \
+                              -t fast-llm-test .
+
+      - name: Run Pytest
+        run: |
+          docker run --rm fast-llm-test \
```

Comment:
Does this run in a gpu environment? Otherwise we're not testing much...

Reply:
Nope, no GPUs. Still, things like configuration validation can be tested.
Huggingface tests their text-generation-inference server releases in the cloud. If we are serious about Fast-LLM, we should do that, too, and pay the money for it.

Reply:
I removed the test workflow. I couldn't make it work given the tight resources in the GitHub action runner.